src/densities.jl

```diff
@@ -50,18 +52,21 @@ grid `basis`, where the individual k-points are occupied according to `occupatio
 
     _check_positive(ρ)
     n_elec_check = weighted_ksum(basis, sum.(occupation))
-    _check_total_charge(basis.dvol, ρ, n_elec_check)
+    _check_total_charge(basis.dvol, ρ, n_elec_check; tol=occupation_threshold)
```

Comment:
Just out of interest: Is this needed? I mean how large is the effect of using the `occupation_threshold` at all?

Reply:
Actually, looking at the output you have now I would revert my opinion, better to use the `occupation_threshold` for the `tol` and avoid the (spurious) warnings. If a user selects a too high occupation threshold than they cannot expect the density to be correct to 1e-10.